auto_scan.py

```diff
@@ -25,43 +26,59 @@
 PAGINATION_MAX = 5000
 WORKER_THREADS = 10
 
+logging.basicConfig(
+    format='%(asctime)s %(name)s [%(levelname)s] %(message)s'
+)
+logger = logging.getLogger('auto-scan')
+logger.setLevel(os.getenv('LOG_LEVEL', logging.INFO))
+
 
 def build_container_assessment_cache(lw_client, start_time, end_time):
-    print(f'Fetching container assessments since "{start_time}"...')
+    logger.info(f'Fetching container assessments since "{start_time}"...')
 
     scanned_containers = lw_client.vulnerabilities.get_container_assessments_by_date(
         start_time=start_time,
         end_time=end_time
     )
 
-    num_returned = len(scanned_containers.get('data', {}))
-    if num_returned == PAGINATION_MAX:
-        logging.warning(f'Warning! The maximum number of container assessments ({PAGINATION_MAX}) was returned.')
-    print(f'Previously Assessed Container Count: {num_returned}')
-
-    print('Building container assessment cache...')
+    logger.info('Building container assessment cache...')
 
+    returned_containers = 0
     scanned_container_cache = {}
 
-    for scanned_container in scanned_containers.get('data', {}):
-        if scanned_container['IMAGE_REPO'] not in scanned_container_cache.keys():
-            scanned_container_cache[scanned_container['IMAGE_REPO']] = scanned_container['IMAGE_TAGS']
+    for scanned_container in scanned_containers.get('data', []):
+
+        registry = scanned_container['IMAGE_REGISTRY']
+        repository = scanned_container['IMAGE_REPO']
+        tags = scanned_container['IMAGE_TAGS']
+
+        qualified_repo = f'{registry}/{repository}'
+
+        if qualified_repo not in scanned_container_cache.keys():
+            scanned_container_cache[qualified_repo] = tags
+            returned_containers += len(tags)
         else:
-            scanned_container_cache[scanned_container['IMAGE_REPO']] += scanned_container['IMAGE_TAGS']
+            for tag in tags:
+                if tag not in scanned_container_cache[qualified_repo]:
+                    scanned_container_cache[qualified_repo].append(tag)
+                    returned_containers += 1
+
+    logger.info(f'Previously Assessed Container Count: {returned_containers}')
+    logger.debug(json.dumps(scanned_container_cache, indent=4))
 
     return scanned_container_cache
 
 
 def build_container_query(registry=None):
 
-    query_text = 'ContainersByRegistry { source { LW_HE_CONTAINERS }'
+    query_text = '{ source { LW_HE_CONTAINERS }'
```

Comment:
Is declaration of a query name no longer required for implicit queries??

Reply:
Yep, no longer required!